### PR TITLE
PHPC-2398: Use server_id methods for libmongoc 1.28+

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-2398

Bumps libmongoc to 1.28-dev

### CODE REVIEW

File: .evergreen/config/templates/build/build-libmongoc.yml [hunk 1]
```diff
@@ -11,7 +11,8 @@ tasks:
       - func: "compile driver"
         vars:
           PHP_VERSION: "%phpVersion%"
-          LIBMONGOC_VERSION: "1.27.2"
+          # TODO: Change to "1.28.0" once 1.28.0 is released
+          LIBMONGOC_VERSION: "master"
```
Comment (alcaeus) [inline .evergreen/config/templates/build/build-libmongoc.yml:15]: This necessitates regenerating the actual build configs.

**Review by alcaeus — CHANGES_REQUESTED** [review]
Evergreen config needs regenerating.

Note: I don't think we need to update the SBOM for this change, especially considering that I forgot how nasty the development version info is. It would need some parsing and/or different handling in order to produce a proper purl.

**Comment by jmikola** [comment]
> Note: I don't think we need to update the SBOM for this change, especially considering that I forgot how nasty the development version info is. It would need some parsing and/or different handling in order to produce a proper purl.

Agreed. I skipped that step from [CONTRIBUTING.md](https://github.com/mongodb/mongo-php-driver/blob/master/CONTRIBUTING.md#updating-bundled-libraries) and think it makes sense to do so provided we never cut a stable release on a dev version of libmongoc.

**Comment by jmikola** [comment]
Evergreen test failures are related to [PHPC-2401](https://jira.mongodb.org/browse/PHPC-2401), since [CDRIVER-4828](https://jira.mongodb.org/browse/CDRIVER-4828) has already been completed in libmongoc 1.28-dev. I'll look into this while this PR is open so we can avoid build failures after merging.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**